### DISCUSSION TITLE
Running Migrations via AdonisJS framework

### DISCUSSION THREAD

**Discussion by adegbengaagoro** [planetscale/discussion#439] [Q&A]
My project is built using AdonisJS v5 and I have successfully established a connection between my project and PlanetScale but when I attempt to run my database migrations, I get this error

<img width="947" alt="Screenshot 2023-04-14 at 21 20 36" src="https://user-images.githubusercontent.com/5008453/232147256-cdcb2db5-4ee5-4ab3-b13a-1a2b2f7f4d65.png">

If anyone has been able to resolve this situation, I would appreciate any insights you can share.

Thank you

**Comment by Reptiluka** [comment]
Same problem here, please I want to use PlanetScale with AdonisJs app :)

**Comment by thetutlage** [comment]
This (https://github.com/planetscale/discussion/discussions/307) is the underlying issue for the error shared by @adegbengaagoro 

We are working on a work-around here https://github.com/adonisjs/lucid/pull/932. You can expect it to work after next release of Lucid

**Reply by orware** [reply]
This is an awesome bit of news @thetutlage about an option being available for your users now to be able to skip locking as one workaround approach!

I'll also refer over to this issue as well https://github.com/adonisjs/lucid/issues/914 I wrote up a few months back as I still think the locking options could potentially be used still, it's just that aliasing them doesn't work yet at our Vitess layer so an alternative approach for checking the result value would be needed.

But with the current workaround implemented by https://github.com/adonisjs/lucid/pull/932 above that at least helps to unblock your users that have been encountering this particular issue ❤️.
